cooked-validators/src/Cooked/MockChain/Monad/GenerateTx.hs

```diff
@@ -36,15 +36,34 @@ generateTxBodyContent,
 generateTxBodyContentWithoutInputDatums = generateTxBodyContent' False
 generateTxBodyContent = generateTxBodyContent' True
 
+-- | Translate a 'TxSkel' into a corresponding 'C.TxBodyContent'.
 generateTxBodyContent' ::
+  -- | Whether to include data on the inputs: Transaction inputs are represented
+  -- on the 'C.TxBodyContent' as a pair of a 'C.TxIn' and a _witness_, which
+  -- records information on how the input is spent. For script inputs there are
+  -- two options with regard to datums: To explicitly include them in the
+  -- witness with the 'C.ScritptDatumForTxIn' constructor, or to leave them
+  -- implicit with the 'C.InlineScriptDatum' constructor. This is what this flag
+  -- chooses.
+  --
+  -- The latter option will (probably, We've not yet completely understood how
+  -- this works!) rely on the information in the UTxO what will be included when
+  -- the 'C.TxBodyContent' is finally transformed into an actual 'C.Tx'.
```

Comment:
Now I'm not sure if I understand this either, but that's probably worth a separate discussion.